packages/turf-intersect/index.ts

```diff
@@ -34,24 +39,29 @@ import polygonClipping from "polygon-clipping";
  *   [-122.520217, 45.535693]
  * ]]);
  *
- * var intersection = turf.intersect(poly1, poly2);
+ * var intersection = turf.intersect(turf.featureCollection([poly1, poly2]));
  *
  * //addToMap
  * var addToMap = [poly1, poly2, intersection];
  */
 export default function intersect<P = GeoJsonProperties>(
-  poly1: Feature<Polygon | MultiPolygon> | Polygon | MultiPolygon,
-  poly2: Feature<Polygon | MultiPolygon> | Polygon | MultiPolygon,
+  features: FeatureCollection<Polygon | MultiPolygon>,
```

Comment:
I see, so by narrowing to a single input parameter that's a FeatureCollection, as opposed to an array in your last PR, we lose the ability to accept simple Geometries, and the caller has to wrap.  Your change to hex-grid is an example.  Gives me mixed feelings.

What if we accept a FC or an array?  Like:
```
features: FeatureCollection<Polygon | MultiPolygon> | Feature<Polygon | MultiPolygon[] | (Polygon | MultiPolygon)[]
```
If `geomEach` is being used in the functions, perhaps it would "just work"?  Food for thought.  I'm not going to block on it though.

Reply:
I guess technically, an array of geometries become just a GeometryCollection in this case.  And well if you have a Feature[] I guess the established pattern with many turf methods is to just accept a FeatureCollection, so I get that argument.

Reply:
For some reason GeometryCollection isn't generic, so I'm hesitant to change this. `geomEach` does not accept arrays in its current for. We can also widen the parameter's type after we release 7.0 without breaking compatibility. I'm also not sure I totally understand your use case, any chance you want to make a PR to better suit your usecase? I'm going to try to get this merged as-is today, but I'm happy to follow up.